benchmarks/src/jmh/java/io/grpc/context/WriteBenchmark.java

```diff
@@ -0,0 +1,86 @@
+package io.grpc.context;
+
+import com.google.common.collect.Lists;
+import io.grpc.Context;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.infra.Blackhole;
+
+/** Write benchmark. */
+public class WriteBenchmark {
+  private static final int NUM_KEYS = 8;
+
+  @State(Scope.Benchmark)
+  public static class ContextState {
+    @Param({"true", "false"})
+    public boolean batched;
+
+    List<Context.Key<Object>> keys = new ArrayList<Context.Key<Object>>();
+    List<List<Context.Key<Object>>> batchKeys = new ArrayList<List<Context.Key<Object>>>();
+    Context context = Context.ROOT;
+    Object val = new Object();
+
+    @Setup
+    public void setup() {
+      for (int i = 0; i < 8; i++) {
+        keys.add(Context.key("Key" + i));
+      }
+      batchKeys = Lists.partition(keys, 4);
+    }
+  }
+
+  @Benchmark
+  @BenchmarkMode(Mode.SampleTime)
+  @OutputTimeUnit(TimeUnit.NANOSECONDS)
+  public void testContextWrite(ContextState state, Blackhole bh) {
+    // always reset Context to avoid unbounded memory usage
+    state.context = Context.ROOT;
+    if (state.batched) {
+      for (Context.Key<Object> key : state.keys) {
```

Comment:
I still find this looping strange, in that they are benchmarking something that doesn't quite seem to matter. It's seem much more valuable to know the cost of modification with a pre-existing context that has a chain of length X.

I feel like setup() should create the context and then this method simply modifies it once. That looks _very_ similar to ReadBenchmark; I'd expect one class that prepares a Context chain in @Setup and has a couple different methods to check the performance of operations with that chain.